Load wallet code dynamically in separate chunks

artursapek opened the pull request:
## Changes

- Disable the Connect Wallet button until the user has selected a chain/network
- Refactor `utils/wallet` so that all chain-specific wallet code lives in separate files, loaded dynamically on demand using `import()`.

Here's a build before these changes:

![image](https://github.com/wormhole-foundation/wormhole-connect/assets/897596/ddca23ed-eef9-4b71-90fb-00952ccb1fd0)

And after:

![image](https://github.com/wormhole-foundation/wormhole-connect/assets/897596/fffc010b-080e-4249-bed6-e6b19d9f2bda)

Our main entry point bundle has gone from 4.51MB to 2.4MB gzipped. There are now several other chunks, each corresponding to a different chain type. These will only be loaded when users actually select that chain and go to connect a wallet.

See it in action here. Upon first loading the page, we load the 15.4 MB (uncompressed) main bundle. Then, if we select Sei chain in the UI and click on Connect Wallet, the page loads the second, 3.5MB uncompressed Sei bundle!

https://github.com/wormhole-foundation/wormhole-connect/assets/897596/879346d9-0372-4df4-8f04-d2348667694f

## Further improvements

- The next low hanging fruit I see is `routes/operator`. Each route that `getRoute` returns could be loaded dynamically. I think this would make worthwhile reductions in main bundle size, because `cosmjs` continues to take up a lot of room in there. I imagine we can get under 2MB after doing that.


## Other thoughts

Besides bundle size/bandwidth savings, this has the added benefit of limiting our attack surface area in vendor code. If one of these wallet libraries gets compromised, we won't be automatically loading and running it for every Connect user.

## TODO

- [x] Add tooltip to disabled connect wallet button

## Follow-up:

https://github.com/wormhole-foundation/wormhole-connect/issues/1616

artursapek committed:
isolate wallet code into separate bundles

artursapek commented on wormhole-connect/src/utils/wallet/index.ts:
```diff
@@ -7,14 +7,15 @@ import {
   MetamaskWallet,
   WalletConnectWallet,
 } from '@xlabs-libs/wallet-aggregator-evm';
+import { SeiTransaction } from '@xlabs-libs/wallet-aggregator-sei';
```
I still have to import this type from `wallet-aggregator-sei`, but it properly tree shakes and avoids bloating the main bundle size.

artursapek commented:
pushed error handling logic, which will tell the user to refresh if a chunk fails to load.

when Connect is properly deployed, this shouldn't happen. but shit happens. or maybe their wifi died.

https://github.com/wormhole-foundation/wormhole-connect/assets/897596/bf834d2a-8da4-4b50-b811-764cf597fb1e

artursapek commented on wormhole-connect/src/utils/wallet.ts:
```diff
@@ -1,409 +0,0 @@
-import {
```
this file was moved to `utils/wallet/index.ts`, and then modified there.

artursapek committed:
show tooltip on disabled "Connect Wallet" button

kev1n-peters commented:
Tested an automatic transfer between Solana and Sui and saw the corresponding wallet code chunks loaded when selecting a wallet. The wallet code splitting is pretty cool.

artursapek committed:
fix mislabeled wallets

artursapek committed:
put OKX Wallet in the right spot

artursapek committed:
clear wallets after bridging

connect clears the chains when this happens, so we should
clear the wallets too

artursapek committed:
move clearWallets to after we leave redeem
